### ISSUE TITLE
HoloMaps and mpld3

### ISSUE THREAD

**Issue by vascotenner** [ioam/holoviews#230]
Today I tried to use the mpld3 backend.  Zooming around in Raster objects works very well. Now I tried to show a HoloMap, but then it complains:
```
%output backend='matplotlib:mpld3'
hv.HoloMap({1:hv.Image(np.random.randint(5, size=(2, 4))),
2: hv.Image(np.random.randint(5, size=(2, 4)))})
```

```
AttributeError
'module' object has no attribute 'NumpyEncoder'
```

Using holoview from master, matplotlib 1.4.3 and mpld3 0.2

**Comment by philippjfr** [comment]
Should add a note to the docs somewhere, we depend on the dev version of mpld3. I'm hoping they'll release v0.3 very soon.

**Comment by vascotenner** [comment]
I just installed the dev version of mpld3. This works better.

However, when I try to set the mpld3 backend for a single cell, this works only the first time I run the cell. The second time it complains:
```
%%output backend='matplotlib:mpld3'
hv.Raster(np.random.random_sample(size=(10,20)))
```
```
Error: Option 'png' not in allowed list: html, json, auto
For help with the %output magic, call %output?
```

**Comment by vascotenner** [comment]
Displaying the holomap still results in an error, but a different one:
```AttributeError
'str' object has no attribute 'items'
```

**Comment by jlstevens** [comment]
> However, when I try to set the mpld3 backend for a single cell, this works only the first time I run the cell. The second time it complains.

This is a known bug I need to fix very soon. It is transient error and if you restart the kernel and immediately switch the backend it should work.

**Comment by philippjfr** [comment]
> Displaying the holomap still results in an error, but a different one:

Just fixed that and another bug, using the slider widget with mpld3 should now work again.

**Comment by jlstevens** [comment]
I've pushed some fixes to the output magic (a90c5a0, e51eb12) and Philipp tells me it now works more reliably. Other than seeing mpld3 0.3, I think that resolves most of my current concerns regarding this backend. 

If  Philipp agrees then I think we can now close this issue.

**Comment by philippjfr** [comment]
Yep, I think we've resolved all the issues that were brought up. I'll go ahead and close.
